ecommerce/api.py

```diff
@@ -55,12 +64,24 @@ def get_purchasable_course_run(course_key, user):
             CourseRun,
             edx_course_key=course_key,
             course__program__live=True,
+            course__program__financial_aid_availability=True,
             courseprice__is_valid=True,
         )
     except Http404:
         log.warning("Course run %s is not purchasable", course_key)
         raise
 
+    if not FinancialAid.objects.filter(
+            tier_program__current=True,
+            tier_program__program__course__courserun=course_run,
+            user=user,
+            status__in=FinancialAidStatus.TERMINAL_STATUSES,
+    ).exists():
+        log.warning("Course run %s has no attached financial aid for user %s", course_key, get_social_username(user))
+        raise ValidationError(
+            "Course run {} does not have a current attached financial aid application".format(course_key)
```

Comment:
nit: I could specify "valid" here, considering that there could be one in a different state, but feel free to leave as is